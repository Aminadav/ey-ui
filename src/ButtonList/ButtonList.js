import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import classNames from 'classnames';
import withStyles from '@material-ui/core/styles/withStyles';

/**
 * A container that holds buttons
 */
const ButtonList = ({ children, align, marginTop, classes }) => {
  const className = classNames(classes.root, {
    [classes.alignTop]: align === 'top',
    [classes.alignBottom]: align === 'bottom',
    [classes.alignRight]: align === 'right',
    [classes.marginTop]: marginTop,
  });

  return (
    <Grid item className={className}>
      {children}
    </Grid>
  );
};

ButtonList.propTypes = {
  align: PropTypes.oneOf(['top', 'bottom', 'right']),
  marginTop: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.array, PropTypes.node]).isRequired,
  classes: PropTypes.shape({
    alignTop: PropTypes.string.isRequired,
    alignBottom: PropTypes.string.isRequired,
    marginTop: PropTypes.string.isRequired,
  }).isRequired,
};

ButtonList.defaultProps = {
  align: 'top',
  marginTop: false,
};

const styles = theme => ({
  root: {
    whiteSpace: 'nowrap',
    marginBottom: theme.spacing.unit * 2,
    display: 'flex',
    justifyContent: 'space-between',
    '& > :not(:first-child)': {
      marginLeft: '1rem',
    },
  },
  alignTop: {
    alignItems: 'flex-start',
  },
  alignBottom: {
    alignItems: 'flex-end',
  },
  alignRight: {
    justifyContent: 'flex-end',
  },
  marginTop: {
    marginTop: '1rem',
  },
});

export default withStyles(styles)(ButtonList);
