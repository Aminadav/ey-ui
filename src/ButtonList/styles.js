import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  root: {
    whiteSpace: 'nowrap',
    marginBottom: theme.spacing(2),
    display: 'flex',
    justifyContent: 'space-between',
    '& > :not(:first-child)': {
      marginLeft: '1rem',
    },
  },
  alignTop: {
    alignItems: 'flex-start',
  },
  alignBottom: {
    alignItems: 'flex-end',
  },
  alignRight: {
    justifyContent: 'flex-end',
  },
  marginTop: {
    marginTop: '1rem',
  },
}));

export default useStyles;
