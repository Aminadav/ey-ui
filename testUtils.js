import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import ThemeProvider from '@material-ui/styles/ThemeProvider';

/**
 * Render dom wrapped with material-ui's ThemeProvider.
 */
export function renderWithTheme(ui) {
  return {
    ...render(<ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>),
  };
}

/**
 * Render dom wrapped with multiple dependent context providers.
 */
export function renderWithProviders(ui) {
  return {
    ...render(
      <MemoryRouter initialEntries={['/']}>
        <ThemeProvider theme={createMuiTheme()}>{ui}</ThemeProvider>
      </MemoryRouter>,
    ),
  };
}

export default { renderWithTheme };
