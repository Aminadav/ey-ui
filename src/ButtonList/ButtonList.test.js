import React from 'react';
import { renderWithTheme } from '../../testUtils';
import ButtonList from './ButtonList';

describe('ButtonList-top', () => {
  it('matches the snapshot - alignTop', () => {
    const { container } = renderWithTheme(
      <ButtonList align="top">
        <button type="submit">Test1</button>
        <button type="submit">Test2</button>
        <button type="submit">Test3</button>
      </ButtonList>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('ButtonList-bottom', () => {
  it('matches the snapshot - alignBottom', () => {
    const { container } = renderWithTheme(
      <ButtonList align="bottom">
        <button type="submit">Test1</button>
        <button type="submit">Test2</button>
        <button type="submit">Test3</button>
      </ButtonList>,
    );
    expect(container).toMatchSnapshot();
  });
});
