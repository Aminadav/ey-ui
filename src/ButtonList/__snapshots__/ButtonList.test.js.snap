// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonList-bottom matches the snapshot - alignBottom 1`] = `
<div>
  <div
    class="MuiGrid-item-109 ButtonList-root-103 ButtonList-alignBottom-105"
  >
    <button
      type="submit"
    >
      Test1
    </button>
    <button
      type="submit"
    >
      Test2
    </button>
    <button
      type="submit"
    >
      Test3
    </button>
  </div>
</div>
`;

exports[`ButtonList-top matches the snapshot - alignTop 1`] = `
<div>
  <div
    class="MuiGrid-item-7 ButtonList-root-1 ButtonList-alignTop-2"
  >
    <button
      type="submit"
    >
      Test1
    </button>
    <button
      type="submit"
    >
      Test2
    </button>
    <button
      type="submit"
    >
      Test3
    </button>
  </div>
</div>
`;
